src/internal/m365/collection/exchange/collection.go

```diff
@@ -36,22 +36,29 @@ const (
 	numberOfRetries             = 4
 )
 
-// Collection implements the interface from data.Collection
-// Structure holds data for an Exchange application for a single user
-type Collection struct {
-	user   string
-	stream chan data.Item
-
-	// added is a list of existing item IDs that were added to a container
-	added map[string]struct{}
-	// removed is a list of item IDs that were deleted from, or moved out, of a container
-	removed map[string]struct{}
-
-	getter itemGetterSerializer
+func NewBaseCollection(
```

Comment:
This looks generic enough that we could move it into `/internal/data` to share the functionality between other packages.  Thoughts on making that move now?  Not updating the other packages with it.  Just making the move.

Reply:
can we do that in another PR? We can update the other packages at that time. Right now I'd like to just get a minimal set of changes for the Exchange kopia assisted incrementals while touching as few different packages as possible and kinda-sorta setup things for easier code movement in the future (hence the generic-ness of this struct 😉)

Reply:
fine by me 👍